[テスト] Durak のバウト上限テストが約1.4%(72回に1回)落ちる — アサーションが自身のコメントの頻度と矛盾している

## 背景・現状

`TestDurak_BoutLimitEndsTheGame/a_normal_game_finishes_well_inside_the_limit` が
**約 1.4% の確率で落ちる**。PR #5797 の CI で観測し、ローカルでも再現した。
当該 PR は Durak を一切触っていない（develop 既存の問題）。

```
--- FAIL: TestDurak_BoutLimitEndsTheGame/a_normal_game_finishes_well_inside_the_limit
    Error: "200" is not less than "200"
```

## 原因: アサーションが、同じテストのコメントが書いている頻度と矛盾している

テスト本体（`internal/domain/Durak_test.go:876`）:

```go
// 健全な局は上限に触れない (実測の最大は 78 バウト)。
t.Run("a normal game finishes well inside the limit", func(t *testing.T) {
    for range 200 {
        d.Reset()
        ...
        assert.Less(t, d.GetBoutNumber(), domain.DurakMaxBouts)   // ← 200 局すべてで上限未満を要求
    }
})
```

同じテストの直前のコメント（#5414 の実測）:

> 20 万局に 14 局という頻度なので、1 局ずつ回して踏むのを待つのは現実的でない。

つまり **1 局あたり 7.0e-5 の確率で上限に到達する**ことが分かっているのに、
**ランダムな 200 局すべてが上限未満**であることを要求している。

```
P(200 局のうち少なくとも 1 局が上限に到達) = 1 - (1 - 14/200000)^200 = 0.0139
→ このサブテストは約 72 回に 1 回落ちる
```

`DurakMaxBouts = 200` なので、到達した局は `GetBoutNumber() == 200` となり
`assert.Less(200, 200)` が落ちる。**上限が正しく働いた局を「異常」と判定している**のが実体で、
プロダクションコードは #5415 の意図どおり動いている。

## 提案

「健全な局は上限に触れない」を**確率的な主張として**書き直す。案:

- 上限に到達した局を**許容してカウント**し、200 局中の到達数が実測頻度から見て
  異常でないこと（例: 数局以内）を確かめる
- あるいは到達局を除いて「到達しなかった局のバウト数が十分小さい」ことを確かめる
- 決定的にしたいなら、シード固定の配りで回す

**単に `assert.Less` を `assert.LessOrEqual` にするだけでは不十分**で、それだと
「上限を超えない」しか言っておらず、元の意図（健全な局は上限から遠い）が消える。

## 受け入れ条件

- [ ] `-count=300` 程度でも安定して緑（現状は約 72 回に 1 回落ちる）
- [ ] 「健全な局は上限から十分離れている」という元の意図が残っている
      （`LessOrEqual` への緩和だけで済ませない）
- [ ] 上限に到達した局が出ても、それが**正しい打ち切り**であることを検証している
      （#5415 の意図が壊れていないこと）
- [ ] 変更後のテストが、上限のロジックを壊したときに落ちること（負のコントロール）

## 対象ファイル

- `internal/domain/Durak_test.go`

---
_PR #5797 の CI で観測。flake-ledger に記録し「1 件目は実バグとして調査」した結果、
プロダクションではなく**テストのアサーションが自分のコメントと矛盾している**と判明したもの。
関連: #5414 / #5415。_

--- a/internal/domain/Durak_test.go
+++ b/internal/domain/Durak_test.go
@@ -872,7 +872,7 @@ func TestDurak_BoutLimitEndsTheGame(t *testing.T) {
 	}
 	d := domain.NewDurak(domain.NewTrumpCardsShortDeck(), players)
 
-	// 健全な局は上限から遠い (実測の最大は 78 バウト)。
+	// 健全な局は上限から遠い (2026-08-16 実測: 20,000 局での最大は 30 バウト)。
 	//
 	// **「200 局すべてが上限未満」は書けない。** すぐ上のコメントが書いているとおり
 	// 20 万局に 14 局は上限に到達するので、200 局を回すと
@@ -895,10 +895,12 @@ func TestDurak_BoutLimitEndsTheGame(t *testing.T) {
 			require.True(t, d.GetGameEndFlag(), "局が終わっていない")
 
 			if d.GetBoutNumber() >= domain.DurakMaxBouts {
+				// 上限に達した局が終局していることは、上の require.True が
+				// 全局について既に保証している。ここで assert.True を重ねても
+				// 決して落ちない空のアサーションになるだけなので置かない。
+				// 打ち切り局が敗者を決めることは "every finished game names a
+				// loser" が別に見ている。
 				capped++
-				// 打ち切られた局も敗者を決めて終わること。ここが崩れると
-				// 「上限に達したので投げ出した」状態を見逃す。
-				assert.True(t, d.GetGameEndFlag(), "上限に達した局が終局していない")
 				continue
 			}
 			if d.GetBoutNumber() > worstNormal {

--- a/internal/domain/Durak_test.go
+++ b/internal/domain/Durak_test.go
@@ -872,18 +872,53 @@ func TestDurak_BoutLimitEndsTheGame(t *testing.T) {
 	}
 	d := domain.NewDurak(domain.NewTrumpCardsShortDeck(), players)
 
-	// 健全な局は上限に触れない (実測の最大は 78 バウト)。
+	// 健全な局は上限から遠い (実測の最大は 78 バウト)。
+	//
+	// **「200 局すべてが上限未満」は書けない。** すぐ上のコメントが書いているとおり
+	// 20 万局に 14 局は上限に到達するので、200 局を回すと
+	// 1 - (1 - 14/200000)^200 = 1.4%、およそ 72 回に 1 回このサブテストが落ちる。
+	// 実際 #5799 として無関係な PR の CI を繰り返し赤くしていた。
+	//
+	// 到達した局は異常ではなく、#5415 で入れた打ち切りが正しく働いた局である。
+	// なので到達を許容しつつ、(1) 到達は稀であること (2) 到達しなかった局は上限の
+	// はるか下で終わること (3) 到達した局もちゃんと終局していること、を確かめる。
 	t.Run("a normal game finishes well inside the limit", func(t *testing.T) {
-		for range 200 {
+		const games = 200
+		capped, worstNormal := 0, 0
+		for range games {
 			d.Reset()
 			turns := 0
 			for !d.GetGameEndFlag() && turns < 500 {
 				d.CpuPlay()
 				turns++
 			}
-			assert.True(t, d.GetGameEndFlag())
-			assert.Less(t, d.GetBoutNumber(), domain.DurakMaxBouts)
+			require.True(t, d.GetGameEndFlag(), "局が終わっていない")
+
+			if d.GetBoutNumber() >= domain.DurakMaxBouts {
+				capped++
+				// 打ち切られた局も敗者を決めて終わること。ここが崩れると
+				// 「上限に達したので投げ出した」状態を見逃す。
+				assert.True(t, d.GetGameEndFlag(), "上限に達した局が終局していない")
+				continue
+			}
+			if d.GetBoutNumber() > worstNormal {
+				worstNormal = d.GetBoutNumber()
+			}
 		}
+
+		// 境界は推測せず実測から決めた (2026-08-16)。この 200 局ループ自体を 300 回
+		// 標本化したところ「200 局中の最大バウト数」は min=21 / p50=25 / p99=29 /
+		// max=31 と狭く分布する。45 なら観測最大の 31 から 45% の余裕があるので
+		// 普通の揺れでは落ちず、バウト数が 2 倍になる退行 (25 -> 50) は捕まえられる。
+		//
+		// 20,000 局での最大も 30 で、コメントにあった「実測の最大は 78」とは合わない。
+		// 78 が何を測った値かは追えなかったので、ここは自分で測った値を根拠にする。
+		const normalBoutCeiling = 45
+		assert.Less(t, worstNormal, normalBoutCeiling,
+			"上限に達しなかった局が長くなりすぎている (2026-08-16 の実測は最大 30 バウト)")
+		// 14/200000 なら 200 局での期待値は 0.014 件。3 件も出るなら頻度が桁違いに増えている。
+		assert.LessOrEqual(t, capped, 3,
+			"上限に達する局が想定より多い (#5414 の実測は 20 万局に 14 局)")
 	})
 
 	// 上限は健全な局の 2 倍以上離れていること。近すぎると普通の局を打ち切る。

